testing/simulated/utils.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/berachain/beacon-kit/primitives/version"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ory/dockertest"
+	"github.com/stretchr/testify/require"
+)
+
+// SharedAccessors holds references to common utilities required in tests.
+type SharedAccessors struct {
+	Ctx        context.Context
+	CancelFunc context.CancelFunc
+	HomeDir    string
+	TestNode   TestNode
+
+	// ElHandle is a dockertest resource handle that should be closed in teardown.
+	ElHandle *dockertest.Resource
+}
+
+// GetBlsSigner returns a new BLSSigner using the configuration files in the provided home directory.
+func GetBlsSigner(tempHomeDir string) *signer.BLSSigner {
+	privValKeyFile := filepath.Join(tempHomeDir, "config", "priv_validator_key.json")
+	privValStateFile := filepath.Join(tempHomeDir, "data", "priv_validator_state.json")
+	return signer.NewBLSSigner(privValKeyFile, privValStateFile)
+}
+
+// CreateInvalidBlock creates a malicious beacon block by injecting an invalid transaction
+// into the execution payload. The invalidity stems from the transaction coming from an account
+// with fee below base fee.
+func CreateInvalidBlock(
+	t *require.Assertions,
+	signedBeaconBlock *ctypes.SignedBeaconBlock,
+	blsSigner *signer.BLSSigner,
+	chainSpec chain.Spec,
+	genesisValidatorsRoot common.Root,
+) *ctypes.SignedBeaconBlock {
+	// Get the current fork version from the slot.
+	forkVersion := chainSpec.ActiveForkVersionForSlot(signedBeaconBlock.GetMessage().Slot)
+
+	// Create a test key - copied from go-ethereum.
+	testKey, err := crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	t.NoError(err, "failed to create test key for malicious transaction")
+
+	// Sign a malicious transaction that is expected to fail.
+	maliciousTx, err := gethtypes.SignNewTx(
+		testKey,
+		gethtypes.NewCancunSigner(big.NewInt(int64(chainSpec.DepositEth1ChainID()))),
+		&gethtypes.DynamicFeeTx{
+			Nonce:     0,
+			To:        &gethcommon.Address{1},
+			Value:     big.NewInt(100000000000),
+			Gas:       100,
+			GasTipCap: big.NewInt(10000000),
+			GasFeeCap: big.NewInt(10000000),
+			Data:      []byte{},
+		},
+	)
+	t.NoError(err, "failed to sign malicious transaction")
+
+	// Initialize the slice with the malicious transaction.
+	maliciousTxs := []*gethprimitives.Transaction{maliciousTx}
+
+	// Extract the current execution payload and compute the withdrawals hash.
+	payload := signedBeaconBlock.GetMessage().GetBody().ExecutionPayload
+	withdrawals := payload.GetWithdrawals()
+	withdrawalsHash := gethprimitives.DeriveSha(withdrawals, gethprimitives.NewStackTrie(nil))
+	parentRoot := signedBeaconBlock.GetMessage().GetParentBlockRoot()
+
+	// Construct a new execution block header and body using the malicious transactions.
+	executionBlock := gethprimitives.NewBlockWithHeader(
```

Comment:
@rezbera I believe I have refactored out this code, so you can reuse `MakeEthBlock` here